File: utils/redis_cache.py
from config.redis import redis


async def cache_checksum(file_name: str, checksum: str):
    try:
        return await redis().set(f"checksum:{file_name}", checksum)
    except Exception as e:
        print(f"Error caching checksum: {e}")
        return None


async def get_cached_checksum(file_name: str):
    try:
        return await redis().get(f"checksum:{file_name}")
    except Exception as e:
        print(f"Error retriving cached checksum: {e}")
        return None

File: src/utils/redisCache.ts
import { redis } from "../config/redis";

/**
 * Caches a checksum value for a given file name
 *
 * @param fileName Name of the file to cache checksum for
 * @param checksum The checksum value to cache
 * @returns Promise resolving to true if successful, null if failed
 */
export const cacheChecksum = async (fileName: string, checksum: string) => {
  try {
    return redis.set(`checksum:${fileName}`, checksum);
  } catch (error) {
    console.error(`Error caching checksum: ${error}`);
    return null;
  }
};

/**
 * Retrieves a cached checksum value for a given file name
 *
 * @param fileName Name of the file to get checksum for
 * @returns Promise resolving to the cached checksum string or null if not found/error
 */
export const getCachedChecksum = async (fileName: string) => {
  try {
    return redis.get<string>(`checksum:${fileName}`);
  } catch (error) {
    console.error(`Error retrieving cached checksum: ${error}`);
    return null;
  }
};
